core/src/main/scala/cats/instances/vector.scala

```diff
@@ -130,6 +130,13 @@ trait VectorInstances extends cats.kernel.instances.VectorInstances {
       final override def traverse[G[_], A, B](fa: Vector[A])(f: A => G[B])(implicit G: Applicative[G]): G[Vector[B]] =
         G.map(Chain.traverseViaChain(fa)(f))(_.toVector)
 
+      final override def updated_[A, B >: A](fa: Vector[A], idx: Long, b: B): Option[Vector[B]] =
+        if (idx >= Int.MinValue.toLong && idx <= Int.MaxValue.toLong) {
+          scala.util.Try(fa.updated(idx.toInt, b)).toOption
```

Comment:
Exceptions can be expensive, meanwhile checking the size of a vector seems to be pretty cheap. So maybe we can explicitly return `None` if `idx` is out of the range of the `Vector`. I'm still skeptical if `Vector` will work at all with more than `Int.MaxValue` elements anyway 🤔

Reply:
I also realized that the check for `Int.MinValue` is redundant as well 😅

Reply:
2.13.8
```
scala> Vector.fill(Int.MaxValue)("")
java.lang.IllegalArgumentException: advance1(1073741824, 2147483616): a1=[Ljava.lang.Object;@38a38ed4, a2=[[Ljava.lang.Object;@1e29a81a, a3=[[[Ljava.lang.Object;@7b4b8199, a4=[[[[Ljava.lang.Object;@4bbf20d1, a5=[[[[[Ljava.lang.Object;@6fced25c, a6=[[[[[[Ljava.lang.Object;@3cf4dec7, depth=6
  at scala.collection.immutable.VectorBuilder.advance1(Vector.scala:1618)
  at scala.collection.immutable.VectorBuilder.advance(Vector.scala:1573)
  at scala.collection.immutable.VectorBuilder.addOne(Vector.scala:1524)
  at scala.collection.immutable.VectorBuilder.addOne(Vector.scala:1349)
  at scala.collection.StrictOptimizedSeqFactory.fill(Factory.scala:330)
  at scala.collection.StrictOptimizedSeqFactory.fill$(Factory.scala:325)
  at scala.collection.immutable.Vector$.fill(Vector.scala:34)
  ... 32 elided
```

Still waiting to see what 2.12 does 😂

Reply:
So far, I've been unable to construct a vector that big. In scala 2.13 it gets the error above. Haven't had any luck with 2.12, the JVM just gets killed no matter how much memory I throw at it.

Reply:
```
$ scala-cli repl -S 2.12 -J -Xms48G -J -Xmx48G
Welcome to Scala 2.12.16 (OpenJDK 64-Bit Server VM, Java 11.0.15).
Type in expressions for evaluation. Or try :help.

scala> val one = Vector("")
one: scala.collection.immutable.Vector[String] = Vector("")

scala> def f(i: Int): Vector[String] = if (i == 0) one else {val z = f(i-1); z ++ z}
f: (i: Int)Vector[String]

scala> f(31).size
Killed
```